File: Leetcode/js/27_removeElement.js
/*
Given an array and a value, 
remove all instances of that value in place and return the new length.

The order of elements can be changed. 
It doesn't matter what you leave beyond the new length.
*/

/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function(nums, val) {
    var crnt = 0;
    
    for(var i=0; i<nums.length; i++){
        if(nums[i] !== val){
            nums[crnt] = nums[i];
            crnt++;
        }
    }
    return crnt;
};

File: Leetcode/java/27_removeElement.java
/*
Given an array and a value, remove all instances of that value in place and return the new length.

The order of elements can be changed. It doesn't matter what you leave beyond the new length.
*/

import java.util.Arrays;

class Main {
    public static void main(String[] args) {
        System.out.println("Starting...");

        Solution solution = new Solution();

        int[] input = {1, 1, 2, 4, 4, 5};

        int k = 1;
        
        int result = solution.removeElement(input, k);
        
        System.out.println("Result: " + result);
        System.out.println("ResultArr: " + Arrays.toString(input));
        
        System.out.println("DONE!");
    }
}

class Solution {
    public int removeElement(int[] nums, int val) {
        int len = nums.length;

        int slow = 0;
        int fast = 0;

        while(fast < len) {
            if(nums[fast] != val) {
                nums[slow] = nums[fast];
                slow++;
            }
            fast++;
        }

        return slow;
    }
}